### PR TITLE
Initializes bindings in connectors and globals

### PR DESCRIPTION
**Pull request by RCHowell**
## Description

This PR introduces a getBindings() method to PartiQL Plugins. A plugin will return a PartiQLBindings object which returns a PartiQLValue given a ConnectorObjectPath.

I have implemented this for the MemoryPlugin which is now used to load globals from the conformance test suite. This PR also includes an implementation of a TestExecutor for the ongoing evaluator work.

- Do your changes comply with the [Contributing Guidelines](https://github.com/partiql/partiql-lang-kotlin/blob/main/CONTRIBUTING.md)
  and [Code Style Guidelines](https://github.com/partiql/partiql-lang-kotlin/blob/main/CODE_STYLE.md)? **[YES/NO]**
Yes

## License Information

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: partiql-eval/src/main/kotlin/org/partiql/eval/internal/Compiler.kt [hunk 1]
```diff
@@ -19,91 +23,129 @@ import org.partiql.plan.Rel
...
-            val constructor = visitRex(node.constructor, ctx)
-            return ExprSelect(rel, constructor)
-        }
+    override fun visitRexOpGlobal(node: Rex.Op.Global, ctx: Unit): Operator {
+        val catalog = plan.catalogs[node.ref.catalog]
+        val symbol = catalog.symbols[node.ref.symbol]
+        val path = ConnectorObjectPath(symbol.path)
+        val bindings = catalogs[catalog.name]!!
+        return ExprGlobal(path, bindings)
+    }
```
Comment (johnedquinn) [inline partiql-eval/src/main/kotlin/org/partiql/eval/internal/Compiler.kt:69]: Glad to see this!

File: plugins/partiql-memory/src/main/kotlin/org/partiql/plugins/memory/MemoryBindings.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    private val bindings: Map<String, PartiQLValue>,
+) : ConnectorBindings {
+
+    @OptIn(PartiQLValueExperimental::class)
+    override fun getValue(path: ConnectorObjectPath): PartiQLValue {
+        val key = path.steps.joinToString(".")
+        return bindings[key] ?: missingValue()
+    }
+
+    internal fun iterator(): Iterator<PartiQLValue> = bindings.values.iterator()
```
Comment (johnedquinn) [inline plugins/partiql-memory/src/main/kotlin/org/partiql/plugins/memory/MemoryBindings.kt:23]: Is this used?

File: plugins/partiql-memory/src/main/kotlin/org/partiql/plugins/memory/MemoryBindings.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+        /**
+         * No bindings.
+         */
+        val empty = MemoryBindings(emptyMap())
+
+        /**
+         * Loads each declared global of the catalog from the data element.
+         */
+        fun load(metadata: MemoryConnector.Metadata, data: StructElement): MemoryBindings {
```
Comment (johnedquinn) [inline plugins/partiql-memory/src/main/kotlin/org/partiql/plugins/memory/MemoryBindings.kt:35]:
```suggestion
        fun load(metadata: MemoryConnector.Metadata, data: Map<String, PartiQLValue>): MemoryBindings {
```
While this is used for the conformance tests, I think this logic would more adequately fit there. The current implementation is specific to the conformance tests.

**Review by johnedquinn — CHANGES_REQUESTED** [review]

File: plugins/partiql-memory/src/main/kotlin/org/partiql/plugins/memory/MemoryBindings.kt (hunk 2 above)
Comment (RCHowell) [reply plugins/partiql-memory/src/main/kotlin/org/partiql/plugins/memory/MemoryBindings.kt:23]: Removed

File: plugins/partiql-memory/src/main/kotlin/org/partiql/plugins/memory/MemoryBindings.kt (hunk 3 above)
Comment (RCHowell) [reply plugins/partiql-memory/src/main/kotlin/org/partiql/plugins/memory/MemoryBindings.kt:35]: Makes sense. I moved `load` to be a private function within the conformance test runner.

**Review by johnedquinn — APPROVED** [review]

**Pull request merged**